jaeger-core/src/main/java/io/jaegertracing/internal/propagation/TextMapCodec.java

```diff
@@ -80,14 +81,29 @@ static JaegerSpanContext contextFromString(String value)
       throw new MalformedTracerStateStringException(value);
     }
 
+    String traceId = parts[0];
+    if (traceId.length() > 32 || traceId.length() < 1) {
+      throw new TraceIdOutOfBoundException("Trace id [" + traceId + "] length is not withing 1 and 32");
+    }
+
     // TODO(isaachier): When we drop Java 1.6 support, use Long.parseUnsignedLong instead of using BigInteger.
     return new JaegerSpanContext(
-        new BigInteger(parts[0], 16).longValue(),
+        high(traceId),
+        new BigInteger(traceId, 16).longValue(),
         new BigInteger(parts[1], 16).longValue(),
         new BigInteger(parts[2], 16).longValue(),
         new BigInteger(parts[3], 16).byteValue());
   }
 
+  private static long high(String hexString) {
```

Comment:
It *looks* like you expect to receive a full id (high + low) in hex here and you are getting only the first 16 chars. If that's indeed the case, please add a javadoc. 

If you are instead expecting only the high to be in this `hexString`, then add also a log message in the condition below (> 16), as it might indicate that the data the tracer is receiving is bad in some way.